pkg/controller/machine/controller.go

```diff
@@ -217,30 +221,12 @@ func (r *ReconcileMachine) Reconcile(ctx context.Context, request reconcile.Requ
 		}
 
 		klog.Infof("%v: reconciling machine triggers delete", machineName)
-		// Drain node before deletion
-		// If a machine is not linked to a node, just delete the machine. Since a node
-		// can be unlinked from a machine when the node goes NotReady and is removed
-		// by cloud controller manager. In that case some machines would never get
-		// deleted without a manual intervention.
-		if _, exists := m.ObjectMeta.Annotations[ExcludeNodeDrainingAnnotation]; !exists && m.Status.NodeRef != nil {
-			// pre-drain.delete lifecycle hook
-			// Return early without error, will requeue if/when the hook owner removes the annotation.
-			if len(m.Spec.LifecycleHooks.PreDrain) > 0 {
-				klog.Infof("%v: not draining machine: lifecycle blocked by pre-drain hook", machineName)
-				return reconcile.Result{}, nil
-			}
-
-			if err := r.drainNode(ctx, m); err != nil {
-				klog.Errorf("%v: failed to drain node for machine: %v", machineName, err)
-				conditions.Set(m, conditions.FalseCondition(
-					machinev1.MachineDrained,
-					machinev1.MachineDrainError,
-					machinev1.ConditionSeverityWarning,
-					"could not drain machine: %v", err,
-				))
-				return delayIfRequeueAfterError(err)
-			}
-			conditions.Set(m, conditions.TrueCondition(machinev1.MachineDrained))
+		// check if machine was already drained
+		drainedCondition := conditions.Get(m, machinev1.MachineDrained)
+		if drainedCondition == nil || drainedCondition.Status != corev1.ConditionTrue {
+			klog.Infof("%v: wait for node draining to proceed deletion", machineName)
```

Comment:
Nit: Minor wording suggestion
```suggestion
			klog.Infof("%s: waiting for node to be drained before deleting instance", machineName)
```